src/uu/kill/src/kill.rs

```diff
@@ -168,32 +173,41 @@ fn print_signals() {
             print!(" ");
         }
     }
+    Ok(())
 }
 
-fn list(arg: Option<String>) {
+fn list(arg: Option<String>) -> UResult<()> {
     match arg {
-        Some(ref x) => print_signal(x),
-        None => print_signals(),
-    };
+        Some(ref x) => Ok(print_signal(x)?),
+        None => Ok(print_signals()?),
+    }
 }
 
-fn kill(signalname: &str, pids: &[String]) -> i32 {
-    let mut status = 0;
+fn kill(signalname: &str, pids: &[String]) -> UResult<()> {
     let optional_signal_value = uucore::signals::signal_by_name_or_value(signalname);
     let signal_value = match optional_signal_value {
         Some(x) => x,
-        None => crash!(EXIT_ERR, "unknown signal name {}", signalname),
+        None => {
+            return Err(USimpleError::new(
+                1,
+                format!("unknown signal name {}", signalname),
+            ));
+        }
     };
     for pid in pids {
         match pid.parse::<usize>() {
             Ok(x) => {
                 if unsafe { libc::kill(x as pid_t, signal_value as c_int) } != 0 {
-                    show_error!("{}", Error::last_os_error());
-                    status = 1;
+                    return Err(USimpleError::new(1, format!("{}", Error::last_os_error())));
```

Comment:
I think this is slightly different than other implementation, because that one doesn't return. I think it should be:
```suggestion
                    show!(USimpleError::new(1, format!("{}", Error::last_os_error())));
```
This shows the error message and sets the exit code for the program.